src/main/java/org/apache/rocketmq/flink/catalog/RocketMQCatalog.java

```diff
@@ -0,0 +1,471 @@
+import org.apache.flink.table.catalog.exceptions.PartitionSpecInvalidException;
+import org.apache.flink.table.catalog.exceptions.TableAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotPartitionedException;
+import org.apache.flink.table.catalog.stats.CatalogColumnStatistics;
+import org.apache.flink.table.catalog.stats.CatalogTableStatistics;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.factories.Factory;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.FieldsDataType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.utils.TypeConversions;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/** Expose a RocketMQ instance as a database catalog. */
+public class RocketMQCatalog extends AbstractCatalog {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RocketMQCatalog.class);
+    public static final String DEFAULT_DB = "default";
+    public final String namesrvAddr;
+    private final String schemaRegistryUrl;
+    private DefaultMQAdminExt mqAdminExt;
+    private SchemaRegistryClient schemaRegistryClient;
+
+    public RocketMQCatalog(
+            String catalogName, String database, String namesrvAddr, String schemaRegistryUrl) {
+        super(catalogName, database);
+        this.namesrvAddr = namesrvAddr;
+        this.schemaRegistryUrl = schemaRegistryUrl;
+        LOG.info("Created RocketMQ Catalog {}", catalogName);
+    }
+
+    @Override
+    public Optional<Factory> getFactory() {
+        return Optional.of(new RocketMQCatalogFactory());
+    }
+
+    @Override
+    public void open() throws CatalogException {
+        if (mqAdminExt == null) {
+            try {
+                mqAdminExt = new DefaultMQAdminExt();
+                mqAdminExt.setNamesrvAddr(namesrvAddr);
+                mqAdminExt.setLanguage(LanguageCode.JAVA);
+                mqAdminExt.start();
+            } catch (MQClientException e) {
+                throw new CatalogException(
+                        "Failed to create RocketMQ admin using :" + namesrvAddr, e);
+            }
+        }
+        if (schemaRegistryClient == null) {
+            schemaRegistryClient = SchemaRegistryClientFactory.newClient(schemaRegistryUrl, null);
+        }
+    }
+
+    @Override
+    public void close() throws CatalogException {
+        if (Objects.nonNull(mqAdminExt)) {
+            mqAdminExt.shutdown();
+        }
+        if (Objects.nonNull(schemaRegistryClient)) {
+            schemaRegistryClient = null;
```

Comment:
Could the `schemaRegistryClient` firstly close and then set to null?

Reply:
`schemaRegistryClient` has not supply `close` or `shutdown` method.